src/moonlink_datafusion/src/connection_pool.rs

```diff
@@ -0,0 +1,276 @@
+use std::{
+    collections::{HashMap, VecDeque},
+    sync::{Arc, LazyLock},
+    time::{Duration, Instant},
+};
+use tokio::{net::UnixStream, sync::Mutex};
+
+const DEFAULT_MAX_ENTRIES_PER_URI: usize = 30;
+const DEFAULT_IDLE_TIMEOUT_MS: u64 = 30_000;
+
+pub(crate) static POOL: LazyLock<Arc<Pool>> = LazyLock::new(|| {
```

Comment:
All connection pool related variables, classes and functions seems to be scoped in the file, which means they could be `private`?